src/clr-angular/data/datagrid/providers/columns.service.ts

```diff
@@ -41,6 +41,37 @@ export class ColumnsService {
     return this.columnStates.filter(state => state.hideable).length > 0;
   }
 
+  get orderOfLastVisible(): number {
+    return Math.max(...this.columnStates.filter(state => !state.hidden).map(state => state.order));
```

Comment:
I like that you kept these calls separate for first/last just in case we have to diverge more in later work.